doc/nrf/app_dev/device_guides/hpf/hpf.rst

```diff
@@ -0,0 +1,24 @@
+.. _hpf_index:
+
+High-Performance Framework (HPF)
+################################
+
+.. note::
+
+   The HPF support in the |NCS| is currently experimental.
+
+High-Performance Framework, is a framework designed to facilitate the creation and integration of software peripherals using coprocessors.
+It includes targeted tools and code examples.
```

Comment:
It should be expanded on general overview of what HPF provides and why VPR is fit for making software peripherals.